pkg/api/validate/openshiftcluster_validatedynamic.go

```diff
@@ -59,66 +60,162 @@ func (dv *openShiftClusterDynamicValidator) Dynamic(ctx context.Context) error {
 	// FP validation
 	fpDynamic, err := dynamic.NewValidator(dv.log, dv.env, dv.env.Environment(), dv.subscriptionDoc.ID, dv.fpAuthorizer, dynamic.AuthorizerFirstParty)
 	if err != nil {
-		return err
+		if _, isCloudError := err.(*api.CloudError); isCloudError {
+			return err
+		}
+		dv.log.Error(err)
+		return api.NewCloudError(
+			http.StatusBadRequest,
+			api.CloudErrorCodeInvalidParameter,
+			"", "Failed to validate the first party principal",
+		)
 	}
 
 	err = fpDynamic.ValidateVnet(ctx, dv.oc.Location, subnets, dv.oc.Properties.NetworkProfile.PodCIDR, dv.oc.Properties.NetworkProfile.ServiceCIDR)
 	if err != nil {
-		return err
+		if _, isCloudError := err.(*api.CloudError); isCloudError {
+			return err
+		}
+		dv.log.Error(err)
+		return api.NewCloudError(
+			http.StatusBadRequest,
+			api.CloudErrorCodeInvalidLinkedVNet,
+			"", "Invalid virtual network specified",
+		)
 	}
 
 	err = fpDynamic.ValidateDiskEncryptionSets(ctx, dv.oc)
 	if err != nil {
-		return err
+		if _, isCloudError := err.(*api.CloudError); isCloudError {
+			return err
+		}
+		dv.log.Error(err)
+		return api.NewCloudError(
+			http.StatusBadRequest,
+			api.CloudErrorCodeInvalidParameter,
+			"", "Invalid disk encryption sets specified",
```

Comment:
This is not very helpful. What exactly is invalid?

Do we miss some specific input which is invalid? If so let's update code inside `ValidateDiskEncryptionSets` to return a specific error. All invalid inputs should be handled inside `ValidateDiskEncryptionSets`.